pan3d/explorers/slice_explorer.py

```diff
@@ -0,0 +1,441 @@
+
+        self.renderer.AddActor(self.outline_actor)
+        self.renderer.AddActor(self.slice_actor)
+        self.renderer.AddActor(self.sbar_actor)
+
+        self.renderer.ResetCamera()
+
+
+    @property
+    def ctrl(self):
+        return self.server.controller
+        
+    @property
+    def state(self):
+        return self.server.state
+
+    @property
+    def t_slice(self):
+        datavar = self.state.data_var
+        m       = self.state.time_active
+        mesh    = self.t_cache.get((m, datavar))
+        if mesh is None:
+            criteria = {}
+            criteria[self.builder.t] = m
+            da = self.builder.dataset[datavar].sel(**criteria)
+            da.load()
+            mesh = da.pyvista.mesh(x=self.builder.x, y=self.builder.y, z=self.builder.z)
+            self.t_cache[(m, datavar)] = mesh
+        return mesh
+ 
+    @property
+    def slice_dimension(self):
+        return self.state.slice_dim
+
+    @slice_dimension.setter
+    def slice_dimension(self, dim : str) -> None:
+        print(f"Setting slice dimension to {dim}")
+        self.state.slice_dim = dim
+        self.UpdateSlicer(dim)
+        self.on_data_change()
+
+    @property
+    def slice_value(self):
+        return self.state.dimval
+
+    @slice_value.setter
+    def slice_value(self, value : float) -> None:
+        print(f"Updating slice value tp {value}")
+        self.state.dimval = value
+        self.on_data_change()
+
+    @property
+    def view_mode(self):
+        return self.state.view_mode
+    
+    @view_mode.setter
+    def view_mode(self, mode):
+        self.state.view_mode = mode
+        self.on_view_mode_change("2D")
+
+    @property
+    def color_map(self):
+        return self.state.cmap
+    
+    @color_map.setter
+    def color_map(self, cmap):
+        self.state.cmap = cmap
+        self.on_colormap_change(cmap)
+
+    @change("cmap")
+    def on_colormap_change(self, cmap, **_):
+        use_preset(self.slice_actor, self.data_actor, cmap)
+        self.ctrl.view_update() 
+
+    @change("logscale")
+    def on_log_scale_change(self, logscale, **_):
+        update_preset(self.slice_actor, logscale)
+        self.ctrl.view_update() 
+
+    @change("view_mode")
+    def on_view_mode_change(self, view_mode, **_):
+        slice_dim = self.state.slice_dim
+        slice_i     = 0 if self.dims[slice_dim] == 'x' else 1 if self.dims[slice_dim] == 'y' else 2
+
+        extents = list(self.extents.values())
+        origin = [
+            float(extents[0][0] + (extents[0][1] - extents[0][0]) / 2), 
+            float(extents[1][0] + (extents[1][1] - extents[1][0]) / 2), 
+            float(extents[2][0] + (extents[2][1] - extents[2][0]) / 2)
+        ]
+
+        camera = self.renderer.GetActiveCamera()
+        pos = camera.GetPosition()
+        norm = np.linalg.norm(np.array(origin) - np.array(pos))
+        print(f"Origin and norm : {origin, norm}")
+        if self.state.view_mode == '3D':
```

Comment:
I would extract that vtk section into helper methods `self._setup_view3d()`, `self._setup_view2d(axis)`

Reply:
```
pos = [*origin]
pos[axis] = norm
view_up = [0,0,1] if axis == 1 else [0,1,0]
camera.SetPosition(pos)
camera.SetViewUp(view_up)
```